CONTRIBUTE.md

```diff
@@ -0,0 +1,108 @@
+
+Good:
+
+> Fixes particle effect of USA Missile Defender
+
+Bad:
+
+> Fixed particle effect of USA Missile Defender
+
+When a text refers to a faction unit, structure, upgrade or similar, then the unit should be worded without any abbrevations and should be prefixed with the faction name. Valid faction names are USA, China, GLA, Boss, Civilian. Subfaction names can be appended too, for example GLA Stealth.
+
+Good:
+
+> Fixes particle effect of USA Missile Defender
+
+Bad:
+
+> Fixes particle effect of MD
+
+
+### 3. Code Documentation
+
+User facing changes need to be accompanied by comment(s) where the change is made. Maintenance related changes, such as compilation fixes, typically do not need commenting, unless the next reader can benefit from a special explanation. The comment can be put at the begin of the changed class, function or block. It must be clear from the change description what has changed.
+
+The expected comment format is
+
+```
+// TheSuperHackers @keyword author DD/MM/YYYY A meaningful description for this change.
+```
+
+The `TheSuperHackers` word and `@keyword` are mandatory. `author` and date can be omitted when preferred.
+
+| Keyword          | Use-case                                                    |
+|------------------|-------------------------------------------------------------|
+| @bugfix          | Fixes a bug                                                 |
+| @fix             | Fixes something, but is not a user facing bug               |
+| @compile         | Addresses a compile warning or error                        |
+| @feature         | Adds something new                                          |
+| @performance     | Improves performance                                        |
+| @refactor        | Moves or rewrites code, but does not change the behaviour   |
+| @tweak           | Changes values or settings                                  |
+| @info            | Writes useful information for the next reader               |
+| @todo            | Adds a note for something left to do if really necessary    |
+
+Block comment sample
+
+```
+    // TheSuperHackers @bugfix JAJames 17/03/2025 Fix uninitialized memory access and add more Windows versions.
+    memset(&os_info,0,sizeof(os_info));
+```
+
+
+### 4. Pull Request Documentation
+
+The title of a new Pull Request, and/or commit(s) within, begin with a **[GEN]** and/or **[ZH]** tag, depending on the game(s) it targets. If a change does not target a game, then a tag is not necessary. Furthermore, the title consists of a concise and descriptive sentence about the change and/or commit, beginning with an uppercase letter and ending without a dot. The title ideally begins with a word that describes the action that the change takes, for example `Fix *this*`, `Change *that*`, `Add *those*`, `Refactor *thing*`.
```

Comment:
Add more information about commit title tag names and the establish names:

[GEN]
[ZH]
[CMAKE]
[GITHUB]
[LINUX]

Reply:
Noted.